energy-integration/farm-boosted-yields/src/lib.rs

```diff
@@ -130,6 +196,64 @@ where
         ManagedVec::from_single_item(EsdtTokenPayment::new(reward_token_id, 0, total_rewards))
     }
 
+    fn get_user_rewards_for_week(
+        &self,
+        module: &Self::WeeklyRewardsSplittingMod,
+        energy_amount: &BigUint<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+        total_energy: &BigUint<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+        total_rewards: &PaymentsVec<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api>,
+    ) -> PaymentsVec<<Self::WeeklyRewardsSplittingMod as ContractBase>::Api> {
+        let mut user_rewards = ManagedVec::new();
+        let factors = module.boosted_yields_factors().get();
+        if energy_amount <= &factors.min_energy_amount
+            || self.user_farm_amount < factors.min_farm_amount
+        {
+            return user_rewards;
+        }
+
+        // user base rewards per week
+        let user_base_rewards_per_block =
+            &self.total_rewards_per_block * &self.user_farm_amount / &self.total_farm_supply;
+        let user_rewards_for_week =
+            &factors.user_rewards_base_const * &user_base_rewards_per_block * BLOCKS_PER_WEEK;
+
+        // computed user rewards
+        // total_boosted_rewards * (energy_const * user_energy / total_energy + farm_const * user_farm / total_farm) / (energy_const + farm_const)
+        for weekly_reward in total_rewards {
+            let boosted_rewards_by_energy =
+                &weekly_reward.amount * &factors.user_rewards_energy_const * energy_amount
+                    / total_energy;
+            let boosted_rewards_by_tokens =
+                &weekly_reward.amount * &factors.user_rewards_farm_const * &self.user_farm_amount
```

Comment:
I think this would be better with self.total_farm_supply_for_those_with_energy, although as users are getting locked MEX rewards - it will be only a small percentage of users who do not have energy at all.